### PR TITLE
NVIDIA driver + Vulkan survey Dec 22'

### PR DESCRIPTION
**Pull request by cthbleachbit**
Topic Description
-----------------

This PR udpates NVIDIA drivers and Vulkan stack to their respective latest versions.

Package(s) Affected
-------------------

- nvidia
- libglvnd, libglvnd+32
- vulkan-headers
- vulkan-loader, vulkan-loader+32
- spirv-headers
- spirv-tools
- vkd3d, vkd3d+32
- wine

Build Order
-----------

- amd64: nvidia libglvnd libglvnd+32 vulkan-headers vulkan-loader vulkan-loader+32 spirv-headers spirv-tools vkd3d vkd3d+32 wine
- arm64: nvidia libglvnd vulkan-headers vulkan-loader spirv-headers spirv-tools vkd3d wine
- riscv64 / loongson3: libglvnd vulkan-headers vulkan-loader spirv-headers spirv-tools

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: extra-libs/spirv-headers/autobuild/beyond [hunk 1]
```diff
@@ -0,0 +1,3 @@
+abinfo "Creating symlink in /opt/32"
+mkdir -p "${PKGDIR}/opt/32/include"
+ln -svf "/usr/include/spirv" "${PKGDIR}/opt/32/include/"
```
Comment (MingcongBai) [inline extra-libs/spirv-headers/autobuild/beyond]: Use relative link here.

File: extra-libs/vulkan-headers/autobuild/defines [hunk 2]
```diff
@@ -7,3 +7,5 @@ ABHOST=noarch
 
 PKGBREAK="vulkan<=1.1.70.1-3"
 PKGREP="vulkan<=1.1.70.1-3"
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-libs/vulkan-headers/autobuild/defines]: Outline why cmake is used in place of cmakeninja here, in a comment.

File: extra-x11/nvidia/spec [hunk 3]
```diff
@@ -16,5 +18,7 @@ CHKSUMS__ARM64="
 	sha256::e3319a184bd1a2813f653691d2c8113e7a79fb25857a3920661d7fa61f6539c3
 	sha256::710a613c6a93a5b90cf48a02530ffd542e4a33b370567bc425d738652b6f87e9
 "
+
 SUBDIR=.
 CHKUPDATE="anitya::id=5454"
+
```
Comment (MingcongBai) [inline extra-x11/nvidia/spec]: Trailing blank line?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by cthbleachbit** [comment]
Note: wine now requires llvm to generate arm64 PE executable and couldn't be build for now. This issue will be corrected in a future topic.

**Pull request merged**